debian/ubuntu-advantage-tools.postinst

```diff
@@ -407,6 +406,51 @@ migrate_ubuntu_pro_beta_banner() {
         fi
     fi
 }
+migrate_old_notices(){
+    notices_file=/var/lib/ubuntu-advantage/notices.json
+    # only run if notices.json is present
+    if [ ! -f $notices_file ];then
+        return
+    fi
+
+    fips_reboot_msg="Reboot to FIPS kernel required"
+    wrong_fips_metapackage_msg_prefix="Warning: FIPS kernel is not optimized for your specific cloud."
+    wrong_fips_metapackage_msg="Warning: FIPS kernel is not optimized for your specific cloud.
+To fix it, run the following commands:
+
+    1. sudo pro disable fips
+    2. sudo apt-get remove ubuntu-fips
+    3. sudo pro enable fips --assume-yes
+    4. sudo reboot
+"
+    fips_system_reboot_msg="FIPS support requires system reboot to complete configuration."
+    livepatch_reboot_msg="Livepatch support requires a system reboot across LTS upgrade."
+
+    if grep -q "$fips_reboot_msg" $notices_file; then
+        filename=/run/ubuntu-advantage/notices/20-fips_reboot_required
+        printf "%s" "$fips_reboot_msg" > $filename
+    fi
+    if grep -q "$wrong_fips_metapackage_msg_prefix" $notices_file; then
+        filename=/var/lib/ubuntu-advantage/notices/25-wrong_fips_metapackage_on_cloud
+        printf "%s" "$wrong_fips_metapackage_msg" > $filename
+    fi
+    if grep -q "System reboot required" $notices_file; then
+        filename=/run/ubuntu-advantage/notices/10-reboot_required
+        printf "%s" "System reboot required" > $filename
+    fi
+    if grep -q "$fips_system_reboot_msg" $notices_file; then
+        filename=/run/ubuntu-advantage/notices/21-fips_system_reboot_required
+        printf "%s" "$fips_system_reboot_msg" > $filename
+    fi
+    if grep -q "$livepatch_reboot_msg" $notices_file; then
+        filename=/run/ubuntu-advantage/notices/30-lp_lts_reboot_required
+        printf "%s" "$livepatch_reboot_msg" > $filename
+    fi
+    #TODO: should we delete the file
+    rm /var/lib/ubuntu-advantage/notices.json
+    # TODO: Should we check for the notices ENABLE_REBOOT_REQUIRED 
```

Comment:
I think we only want to do the simplest possible migration here, since the probability that any of this is needed is already tiny.

Also we should stick with one method of creating notices in the postinst. For now that is the `add_notice` function. https://warthogs.atlassian.net/browse/SC-1358 will change it.

Also yes we should delete the old file.

So in all I think it will look something like:
```suggestion
    # This migration will happen for pro client versions 27.12 or older that still use notices.json.
    # Notices are generally short lived, so the chances of an upgrade happening while a
    # notice is in place is very small.
    # Despite that we do a simple migration here of the most important notices: reboot required notices.
    # All notices with "reboot" in them can be safely transformed into a generic reboot required message.
    # The new message won't include the exact reason for the reboot, but the recommended action is the same.
    if grep -q -i "reboot" $notices_file; then
        add_notice REBOOT_REQUIRED
    fi

    rm $notices_file
```

Reply:
@renanrodrigo @lucasmoura @CalvoM please confirm that this is what we agreed to the other day